sunpy/net/dataretriever/client.py

```diff
@@ -300,6 +300,15 @@ def _get_full_filenames(self, qres, filenames, path):
 
         return paths
 
+    def _get_time_for_url(self, urls):
+        """
+        This method allows clients to customise the timerange displayed for
+        each URL.
+
+        It should return a sunpy.time.TimeRange object per URL.
```

Comment:
a TimeRange? what if you don't know from the URL what's the obs-end?

Reply:
:rofl: you wrote this function, I just added the docstring!

If you don't know the end time, I guess you make it up! It makes no sense to return anything other than a range here, all files will have some range, I would really hope you know what it is.

Reply:
Really?? Then of course we know it!!
I'll have to look into it with more detail... but not know. Let's merge this and one day I'll get my head around the scrapper again.